### PR TITLE
JENA-1639: Model.getList and Statement.getList

### PR DESCRIPTION
**Pull request by afs**
Attempted to use the local code style at each addition point.

### CODE REVIEW

**Comment by ajs6f** [comment]
I'm probably confused, but where is `Model::getList` being added? I just see a change to `createList` and `getList` is only added to `ModelCon`.

File: jena-core/src/main/java/org/apache/jena/rdf/model/ModelCon.java [hunk 1]
```diff
@@ -111,38 +111,62 @@
...
+ * <p> This method enables an application to treat any resource as a list.
+ *     It is in effect an unsafe downcast.</p>
+ *
+ * <p>Subsequent operations on the returned list may modify this model.</p>
+ * <p>The list is assumed to already exist in the model.  If it does not,
+ * <CODE>createList</CODE> should be used instead.</p>
+ * @return a list instance
+ * @param r the resource of the list
+ */ 
+   RDFList getList(Resource r) ;
```
Comment (afs) [inline jena-core/src/main/java/org/apache/jena/rdf/model/ModelCon.java]: Here.

**Comment by afs** [comment]
`interface Model extends ModelCon`

(I think to put many of the convenience wrappings in a different place i.e. `ModelCon`)

**Comment by ajs6f** [comment]
Ah, I _always_  get `ModelCon` and `ModelCom` (impl) confused. Cool!

**Comment by afs** [comment]
Ditto!

**Pull request merged**